modules/nexverseBidAdapter.js

```diff
@@ -0,0 +1,447 @@
+   */
+  isBidRequestValid(bid) {
+    const isValid = !!(bid.params && bid.params.uid && bid.params.pub_id && bid.params.pub_epid);
+
+    if (!isValid) {
+      logError(`${LOG_ERROR_PREFIX} Missing required bid parameters.`);
+    }
+
+    return isValid;
+  },
+
+  /**
+   * Builds the OpenRTB server request from the list of valid bid requests.
+   *
+   * @param {Array} validBidRequests - Array of valid bid requests.
+   * @param {Object} bidderRequest - The bidder request object containing additional data.
+   * @returns {Array} Array of server requests to be sent to the endpoint.
+   */
+  buildRequests(validBidRequests, bidderRequest) {
+    const requests = validBidRequests.map((bid) => {
+      const { uid, pub_id, pub_epid } = bid.params;
+
+      if (!uid || !pub_id || !pub_epid) {
+        logError(`${LOG_ERROR_PREFIX} Missing required endpoint URL parameters.`);
+        return null; // Stop processing this bid
+      }
+
+      // Build the endpoint URL with query parameters
+      const endpointUrl = `${ENDPOINT_URL}?uid=${encodeURIComponent(uid)}&pub_id=${encodeURIComponent(
+        pub_id
+      )}&pub_epid=${encodeURIComponent(pub_epid)}`;
+
+      // Build the OpenRTB payload
+      const payload = buildOpenRtbRequest(bid, bidderRequest);
+
+      if (!payload) {
+        logError(`${LOG_ERROR_PREFIX} Payload could not be built.`);
+        return null; // Skip this bid
+      }
+
+      // Return the server request
+      return {
+        method: 'POST',
+        url: endpointUrl,
+        data: JSON.stringify(payload),
+        bidRequest: bid,
+        options: {
+          contentType: 'application/json',
+          withCredentials: true,
+        },
+      };
+    });
+
+    return requests.filter((request) => request !== null); // Remove null entries
+  },
+
+  /**
+   * Interprets the server's response and extracts bid information.
+   *
+   * @param {Object} serverResponse - The response from the server.
+   * @param {Object} request - The original server request.
+   * @returns {Array} Array of bids to be passed to the auction.
+   */
+  interpretResponse(serverResponse, request) {
+    if (serverResponse && serverResponse.status === 204) {
+      logInfo(`${LOG_INFO_PREFIX} No ad available (204 response).`);
+      return [];
+    }
+
+    const bidResponses = [];
+    const response = serverResponse.body;
+
+    if (!response || !response.seatbid || !isArray(response.seatbid)) {
+      logWarn(`${LOG_WARN_PREFIX} No valid bids in the response.`);
+      return bidResponses;
+    }
+
+    response.seatbid.forEach((seatbid) => {
+      seatbid.bid.forEach((bid) => {
+        const bidResponse = {
+          requestId: bid.impid,
+          cpm: bid.price,
+          currency: response.cur || 'USD',
+          width: bid.w || 0,
+          height: bid.h || 0,
+          creativeId: bid.crid || bid.id,
+          ttl: 300,
+          netRevenue: true,
+          meta: {},
```

Comment:
You must be able to set adomain

Reply:
Fixed in new PR https://github.com/prebid/Prebid.js/pull/12472

Reply:
Done.